Algorithm.Python/CustomIndicatorWithExtensionAlgorithm.py

```diff
@@ -0,0 +1,74 @@
+# QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+# Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from AlgorithmImports import *
+
+from collections import deque
+from math import isclose
+
+class CustomIndicatorWithExtensionAlgorithm(QCAlgorithm):
+    def Initialize(self):
+        self.SetStartDate(2013, 10, 9)
+        self.SetEndDate(2013, 10, 9)
+
+        self.spy = self.AddEquity("SPY", Resolution.Minute).Symbol
+
+        self.sma_values = []
+        self.period = 10
+
+        self.sma = self.SMA(self.spy, self.period, Resolution.Minute)
+        self.sma.Updated += self.OnSMAUpdated
+
+        self.custom_sma = CustomSimpleMovingAverage("My SMA", self.period)
+        self.ext = IndicatorExtensions.Of(self.custom_sma, self.sma)
+        self.ext.Updated += self.OnIndicatorExtensionUpdated
+
+        self.sma_minus_custom = IndicatorExtensions.Minus(self.sma, self.custom_sma)
+        self.sma_minus_custom.Updated += self.OnMinusUpdated
+
+    def OnSMAUpdated(self, sender, updated):
+        if self.sma.IsReady:
+            self.sma_values.append(self.sma.Current.Value)
+
+    def OnIndicatorExtensionUpdated(self, sender, updated):
```

Comment:
Should we assert at the end of the algorithm these callbacks were called at least once to be safe?